share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorService.java

```diff
@@ -91,6 +92,7 @@ public class ShareCoordinatorService implements ShareCoordinator {
     private final Timer timer;
     private final PartitionWriter writer;
     private final Map<TopicPartition, Long> lastPrunedOffsets;
+    private final AtomicBoolean shouldRunPeriodJob = new AtomicBoolean(true);
```

Comment:
Why do you need an AtomicBoolean?

Reply:
onMetadataUpdate called by separate thread (broker metadata publisher)

Reply:
Yeah but why AtomicBoolean for the current implementation and not `volatile boolean`, there are just get/set simple operations in the PR. AtomicBoolean is also a wrapper over `volatile`. Haing said that, if onMetadataUpdate can be called by separate threads and concurrently the instead of `shouldRunPeriodJob.set(enabled);` we should use `shouldRunPeriodJob.compareAndSet(false, true)`, if the operation is successful then only we should trigger the `setupRecordPruning`. Else there can multiple instances of job running.